src/aiida/cmdline/commands/cmd_process.py

```diff
@@ -180,14 +180,17 @@ def process_list(
             echo.echo_report(f'Using {percent_load * 100:.0f}% of the available daemon worker slots.')
 
 
-@verdi_process.command('show')
+@verdi_process.command('show', short_help='Show details for the given process(es).')
 @arguments.PROCESSES()
 @options.MOST_RECENT_NODE()
 @decorators.with_dbenv()
 def process_show(processes, most_recent_node):
-    """Show details for one or multiple processes."""
+    """Show details for one or multiple processes given the primary key(s) (PK)."""
     from aiida.cmdline.utils.common import get_node_info
 
+    if not (processes) and not (most_recent_node):
+        raise click.BadArgumentUsage('Please specify process primary key(s) (PK) or an option.')
```

Comment:
Think `UsageError` might be more appropriate since it really involves the lack of arguments _and_ an option
```suggestion
        raise click.UsageError('Please specify process primary key(s) (PK) or an option.')
```

Reply:
Taking comment on the help message into account of show. What about this message? 
```
Please specify process(es) identifier (PK, UUID or label) or an option.
```

Reply:
Ok, this is fine for the exception

Reply:
Can you actually retrieve processes with label? When I try it, it does not work

Reply:
what did you use for a label, how did you set it on the process node, and what was the exact verdi command?

Reply:
Had some processes from running the basic example (But I don't remember what I exactly have run)
```
(aiida-dev) alexgo@fw:~/code/aiida-core(fix/6391/verdi-process-show)$ verdi process list -a
  PK  Created    Process label             ♻    Process State    Process status
----  ---------  ------------------------  ---  ---------------  ----------------
   5  7D ago     MultiplyAddWorkChain           ⏹ Finished [0]
   6  7D ago     multiply                       ⏹ Finished [0]
   8  7D ago     ArithmeticAddCalculation       ⏹ Finished [0]
  15  7D ago     MultiplyAddWorkChain           ⏹ Finished [0]
  16  7D ago     multiply                       ⏹ Finished [0]
  18  7D ago     ArithmeticAddCalculation       ⏹ Finished [0]

```
then wanted to verify the label name, since in the list they are not unique so I thought `Process label` means something else
```
(aiida-dev) alexgo@fw:~/code/aiida-core(fix/6391/verdi-process-show)$ verdi process show 6
Property     Value
-----------  ------------------------------------
type         multiply
state        Finished [0]
pk           6
uuid         99217f69-1498-4619-906f-ef115522d6fc
label        multiply
description
ctime        2024-05-30 13:49:12.576837+00:00
mtime        2024-05-30 13:49:12.875801+00:00

Inputs      PK  Type
--------  ----  ------
x            2  Int
y            3  Int

Outputs      PK  Type
---------  ----  ------
result        7  Int

Caller      PK  Type
--------  ----  --------------------
CALL         5  MultiplyAddWorkChain
```
In this case the label is still `multiply` which is confusing as it is not unique, so I tried it nevertheless
```
Usage: verdi process show [OPTIONS] [PROCESSES]...
Try 'verdi process show --help' for help.

Error: Invalid value for '[PROCESSES]...': multiple ProcessNode entries found with LABEL<multiply>
```
I guess label is not what I think it is? Or maybe this is a bug because other processes dont have labels, only these `multiply` processes

Reply:
The `Process label` in `verdi process list` is not the `label` that can be used to identify a node. The `label` is actually a column in the database schema and is exposed as a property, i.e., `Node.label`. The process label is stored as the `process_label` attribute and is only defined for `ProcessNode`s.

Regarding the error: this is not a bug, the error message is correct. The `label` is not guaranteed to be unique (unlike the PK and UUID) and this is intentional. So when users want to use labels to be able to identify nodes, it is up to them to guarantee they are unique.

The fact that the `multiple` node actually has `multiple` set as its label surprises me a bit tbh. It seems that the process label may also actually be set automatically as the `label`. I would say that this is not desirable. Will have to check in the code if and where this might be happening